### ISSUE TITLE
Traceback when using the `thumbnail` directive

### ISSUE THREAD

**Issue by e2jk** [getnikola/nikola#1598]
When using the `thumbnail` directive, for an image that is placed in the `images` folder, I get the following Traceback:

```
.  scale_images:output/lightweight-django.jpg
########################################
TaskError - taskid:scale_images:output/lightweight-django.jpg
PythonAction Error
Traceback (most recent call last):
  File "/home/emilien/Envs/nikola/local/lib/python2.7/site-packages/doit/action.py", line 365, in execute
    returned_value = self.py_callable(*self.args, **kwargs)
  File "/home/emilien/Envs/nikola/local/lib/python2.7/site-packages/nikola/plugins/task/scale_images.py", line 68, in process_image
    self.resize_image(src, '.thumbnail'.join(os.path.splitext(dst)), self.kw['thumbnail_size'])
  File "/home/emilien/Envs/nikola/local/lib/python2.7/site-packages/nikola/image_processing.py", line 84, in resize_image
    self.logger.warn("Can't thumbnail {0}, using original "
AttributeError: 'ScaleImage' object has no attribute 'logger'
```

The resulting .html page contains the following:
```
<a class="reference external image-reference" href="../../assets/img/lightweight-django.jpg"><img alt="/assets/img/lightweight-django.thumbnail.jpg" src="../../assets/img/lightweight-django.thumbnail.jpg"></a>
```

It displays a "broken image", but clicking on the link shows the fullscale image.

Any idea what dependency I'm missing for the ScaleImage object to have a 'logger' attribute? Some Pillow dependency that might not be fulfilled?

I have installed Nikola 7.3.0 from `pip`, both packages `nikola` and `nikola[extras]`.

**Comment by Kwpolska** [comment]
> Any idea what dependency I'm missing for the ScaleImage object to have a 'logger' attribute?

nah, that’s our fault, scaleimage tries to use a Nikola logger but doesn’t ask for one.

--- 
Paste `pip freeze` and share `lightweight-django.jpg` somewhere, I’ll take a look later.

**Comment by e2jk** [comment]
The image is [this one](http://akamaicovers.oreilly.com/images/0636920032502/lrg.jpg), saved locally and renamed.

```
$ pip freeze
backports.ssl-match-hostname==3.4.0.2
blinker==1.3
certifi==14.5.14
docutils==0.12
doit==0.26.0
ghp-import==0.4.1
ipython==2.3.1
Jinja2==2.7.3
livereload==2.3.2
Logbook==0.8.1
lxml==3.4.1
Mako==1.0.0
Markdown==2.5.2
MarkupSafe==0.23
micawber==0.3.1
natsort==3.5.2
Nikola==7.3.0
phpserialize==1.3
Pillow==2.7.0
pygal==1.6.1
Pygments==2.0.2
pyinotify==0.9.5
Pyphen==0.9.1
PyRSS2Gen==1.1
python-dateutil==2.4.0
requests==2.5.1
six==1.9.0
smartypants==1.8.6
tornado==4.0.2
typogrify==2.0.7
Unidecode==0.4.17
webassets==0.10.1
Yapsy==1.10.423
```

**Comment by Kwpolska** [comment]
This is caused by:

1. Nikola’s image scaling plugin not having a logger.  This was fixed in 50828ba.
2. The emergency *copy big → thumbnail* fails, because it appears below the warning.
3. Your copy of Pillow seems faulty.  Please fix it.  If you will apply the patch over your copy (which is 3 lines of code), you will see the exact error that caused this issue in your build logs.  You may need to force a rebuild (`nikola build -a scale_images`).

**Comment by e2jk** [comment]
This is the output after applying the patch:

```
t$ nikola build
Scanning posts....done!
.  scale_images:output/lightweight-django.jpg
[2015-01-27T19:57:41Z] WARNING: scale_images: Can't thumbnail images/lightweight-django.jpg, using original image as thumbnail (decoder jpeg not available)
```

Now the process doesn't crash anymore, which is good. The suitable alternative (un-thumbnailed image in the output folder) is used.
Looks like my PIL is not able to handle jpeg, huh. Will investigate and report back, FYI.

**Comment by e2jk** [comment]
This is what solved it (Ubuntu 14.04):

```
$ sudo apt-get install libjpeg-dev
$ pip install Pillow
$ nikola build -a scale_images
Scanning posts....done!
.  scale_images:output/assets/img/lightweight-django.jpg
```

(note that the second command didn't work the first time, I had to use `$ sudo pip install --upgrade pillow` first, then I got the Nikola error

```
[2015-01-27T20:17:03Z] ERROR: Nikola: In order to render galleries, you must install the "pillow" Python package.
[2015-01-27T20:17:03Z] ERROR: Nikola: Exiting due to missing dependencies.
```

but after re-running `pip install Pillow` it worked fine...

Thanks for your time fixing the logger issue.

**Comment by Kwpolska** [comment]
You're welcome!

Then again, finding an issue like this doesn't take long, especially because the traceback tells you everything.
